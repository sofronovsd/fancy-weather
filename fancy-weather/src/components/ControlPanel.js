import React from "react";
import {Button, ButtonGroup, Dropdown, DropdownItem} from "react-bootstrap";

export default function ControlPanel({language, isC, handleRefreshImage, handleChangeC, handleChangeLanguage}) {
    const options = [
        "EN",
        "RU",
        "BE",
    ];

    const handleLanguageSelect = (event) => {
        handleChangeLanguage(event.target.textContent);
    };

    const handleChangeTemp = (event) => {
        handleChangeC(event.target);
    };

    const buttonCClassName = `button ${isC  && 'active'}`;
    const buttonFClassName = `button ${!isC && 'active'}`;

    return (
        <div className="control-panel">
            <Button
                data-testid="button"
                variant="secondary"
                className="button button_refresh"
                onClick={handleRefreshImage}
            >
                <div className="icon-refresh"/>
            </Button>
            <Dropdown data-testid="dropdown">
                <Dropdown.Toggle variant="secondary" className="button button_medium">
                    {language}
                </Dropdown.Toggle>

                <Dropdown.Menu variant="secondary">
                    { options.map((option, index) => {
                        return <DropdownItem key={index} onClick={handleLanguageSelect}>{option}</DropdownItem>
                    }) }
                </Dropdown.Menu>
            </Dropdown>
            <ButtonGroup data-testid="optionButtons">
                <Button
                    data-testid="cButton"
                    variant="secondary"
                    className={buttonCClassName}
                    onClick={handleChangeTemp}
                    data="true"
                >&deg;C</Button>
                <Button
                    data-testid="fButton"
                    variant="secondary"
                    className={buttonFClassName}
                    onClick={handleChangeTemp}
                    data="false"
                >&deg;F</Button>
            </ButtonGroup>
        </div>
    );
}
