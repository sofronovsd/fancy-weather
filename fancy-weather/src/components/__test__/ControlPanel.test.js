import ControlPanel from "../ControlPanel";
import React from "react";
import ReactDOM from "react-dom";
import {render} from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom"

describe('Test control panel functionality', () => {
  it('render control panel without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ControlPanel/>, div)
  });

  it('render control panel correctly', () => {
    const {getByTestId} = render(<ControlPanel/>);
    expect(getByTestId('button')).toBeDefined();
    expect(getByTestId('dropdown')).toBeDefined();
    expect(getByTestId('optionButtons')).toBeDefined();
    expect(getByTestId('cButton')).toBeDefined();
    expect(getByTestId('fButton')).toBeDefined();
  });

  it('render control panel with C degree', () => {
    const {getByTestId} = render(<ControlPanel isC={true}/>);
    expect(getByTestId('optionButtons')).toBeDefined();
    const cButton = getByTestId('cButton');
    expect(cButton).toBeDefined();
    expect(cButton).toHaveClass('active');
  });

  it('render control panel with F degree', () => {
    const {getByTestId} = render(<ControlPanel isC={false}/>);
    expect(getByTestId('optionButtons')).toBeDefined();
    const fButton = getByTestId('fButton');
    expect(fButton).toBeDefined();
    expect(fButton).toHaveClass('active');
  });

  it('control panel matches snapshot', () => {
    const tree = renderer.create(<ControlPanel/>);
    expect(tree).toMatchSnapshot();
  });
});
