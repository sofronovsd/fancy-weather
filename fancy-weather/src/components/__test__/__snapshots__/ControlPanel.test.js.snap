// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test control panel functionality control panel matches snapshot 1`] = `
<div
  className="control-panel"
>
  <button
    className="button button_refresh btn btn-secondary"
    data-testid="button"
    disabled={false}
    type="button"
  >
    <div
      className="icon-refresh"
    />
  </button>
  <div
    className="dropdown"
    data-testid="dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="button button_medium dropdown-toggle btn btn-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    />
  </div>
  <div
    className="btn-group"
    data-testid="optionButtons"
    role="group"
  >
    <button
      className="button undefined btn btn-secondary"
      data="true"
      data-testid="cButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      °C
    </button>
    <button
      className="button active btn btn-secondary"
      data="false"
      data-testid="fButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      °F
    </button>
  </div>
</div>
`;
